archinstall/lib/installer.py

```diff
@@ -178,146 +173,168 @@ def __exit__(self, *args :str, **kwargs :str) -> None:
 			self.sync_log_to_install_medium()
 			return False
 
-	@property
-	def partitions(self) -> List[Partition]:
-		return get_partitions_in_use(self.target).values()
+	def log(self, *args :str, level :int = logging.DEBUG, **kwargs :str):
+		"""
+		installer.log() wraps output.log() mainly to set a default log-level for this install session.
+		Any manual override can be done per log() call.
+		"""
+		log(*args, level=level, **kwargs)
 
-	def sync_log_to_install_medium(self) -> bool:
-		# Copy over the install log (if there is one) to the install medium if
-		# at least the base has been strapped in, otherwise we won't have a filesystem/structure to copy to.
-		if self.helper_flags.get('base-strapped', False) is True:
-			if filename := storage.get('LOG_FILE', None):
-				absolute_logfile = os.path.join(storage.get('LOG_PATH', './'), filename)
+	def _verify_service_stop(self):
+		"""
+		Certain services might be running that affects the system during installation.
+		Currently, only one such service is "reflector.service" which updates /etc/pacman.d/mirrorlist
+		We need to wait for it before we continue since we opted in to use a custom mirror/region.
+		"""
+		log('Waiting for automatic mirror selection (reflector) to complete...', level=logging.INFO)
+		while service_state('reflector') not in ('dead', 'failed'):
+			time.sleep(1)
 
-				if not os.path.isdir(f"{self.target}/{os.path.dirname(absolute_logfile)}"):
-					os.makedirs(f"{self.target}/{os.path.dirname(absolute_logfile)}")
+		log('Waiting pacman-init.service to complete.', level=logging.INFO)
+		while service_state('pacman-init') not in ('dead', 'failed'):
```

Comment:
See 3f5170f

This is the second commit to `master` that I have discovered to not be merged in. Any commit to `master` on code that you have moved to a different or new file may not be successfully merged.